### PR TITLE
Auto monkey return for slime management console.

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request

Auto monkey return for slime management console.
Monkeys returned on it death

monkey spawned > monkey died > monkey sucked
monkey bound to console that spawn it. 
Second console cant take monkeys that it dont spawn.

## Why It's Good For The Game

Less miasma in xeno. 

## Changelog
:cl:
add: The Applied Bluespace Research node now contains an update disk for the Slime management console. It contains a set of microsensors and beacons that allow the console to automatically retrieve and recycle dead monkeys created by the updated console.
/:cl:

Sorry for this upgrade system, that i took from rcd.

### CODE REVIEW

**Comment by optimumtact** [comment]
Power creep.

**Comment by Dennok** [comment]
What if make it upgrade?

**Comment by optimumtact** [comment]
put it on the techweb and I guess it's fine.

**Comment by Bokkiewokkie** [comment]
What about rooms with multiple consoles though, do you just need to deconstruct the other one if you don't want it sucking up 50% of the monkeys?

**Comment by Dennok** [comment]
> What about rooms with multiple consoles though, do you just need to deconstruct the other one if you don't want it sucking up 50% of the monkeys?

monkey spawned > monkey died > monkey sucked
monkey bound to console that spawn it.

**Comment by Bokkiewokkie** [comment]
> monkey bound to console that spawn it.

Ah my bad then, I though it was just looking for (dead) console spawned monkeys in general

**Comment by Dennok** [comment]
> > monkey bound to console that spawn it.
> 
> Ah my bad then, I though it was just looking for (dead) console spawned monkeys in general

I think this way create more load.

**Comment by knacker48** [comment]
This is gonna save a lot of time, although it will make taking a monkey corpse for blood no longer an option. Also I'll probably need to upgrade the recycler a bit earlier as I usually leave the corpses until I get upgrades

**Comment by Dennok** [comment]
Now upgrade console when you want.
It save click hunting. And some time.

**Comment by Rasselbok** [comment]
> Power creep disguised as QoL

Is it really, though? It doesn't actually save any actual time or real work, it just gets rid of minor tedium.  Monkey corpses don't vanish, miasma in xenobio is essentially never harmful since it's almost always inside a pen, and all most people ever do is just pick up dead monkeys as they drop new ones which means that all this QoL change does it change dropping monkeys from clicking on the same tile twice to clicking on the tile once. Where's the power increase? It doesn't decrease complexity or time consumption, it just makes it 'cleaner'.

**Comment by optimumtact** [comment]
Thank you for so clearly describing the creep for me.

**Comment by TheVekter** [comment]
> 
> 
> Power creep disguised as QoL

Not sure if you're joking or not. Don't really see how this is a power creep thing, it saves maybe an extra five seconds and results in less misclicks.

**Comment by optimumtact** [comment]
LGTM now dennok

File: code/modules/research/designs/bluespace_designs.dm [hunk 1]
```diff
@@ -73,3 +73,13 @@
 	build_path = /obj/item/swapper
 	category = list("Bluespace Designs")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE
+
+/datum/design/xeno_up_redemption
+	name = "Automatised Monkey Recycling"
+	desc = "Slime management console upgrade disk. Contain set of micro sensors and beacons allow automatised monkey recycling."
```
Comment (XDTM) [inline code/modules/research/designs/bluespace_designs.dm]:
```suggestion
	desc = "Upgrade disk for the Slime Management Console. Contains a set of micro sensors and beacons that allow the console to automatically retrieve and recycle dead monkeys."

```

File: code/modules/research/designs/bluespace_designs.dm [hunk 2]
```diff
@@ -73,3 +73,13 @@
 	build_path = /obj/item/swapper
 	category = list("Bluespace Designs")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE
+
+/datum/design/xeno_up_redemption
+	name = "Automatised Monkey Recycling"
```
Comment (XDTM) [inline code/modules/research/designs/bluespace_designs.dm]:
```suggestion
	name = "Automated Monkey Recycling"

```

File: code/modules/research/xenobiology/xenobio_camera.dm [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+/obj/item/xeno_console_upgrade
+	name = "Slime management console upgrade disk"
+	desc = "It seems to be empty."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "datadisk3"
+	var/upgrade
+
+///Slime management console redemption upgrade
+/obj/item/xeno_console_upgrade/redemption
+	desc = "Set of micro sensors and beacons allow automatised monkey recycling."
```
Comment (XDTM) [inline code/modules/research/xenobiology/xenobio_camera.dm]:
```suggestion
	desc = "Upgrade disk for the Slime Management Console. Contains a set of micro sensors and beacons that allow the console to automatically retrieve and recycle dead monkeys."
```

**Comment by XDTM** [comment]
Make sure to write in the changelog that it's an upgrade disk and not available by default now

**Comment by Qustinnus** [comment]
lol this is powercreep for no good reason other than "xenobio should do even less"

**Comment by XDTM** [comment]
@Qustinnus It's been changed to a researchable upgrade in mini bluespace (same as brped), although the PR body didn't get updated

**Comment by Rasselbok** [comment]
> lol this is powercreep for no good reason other than "xenobio should do even less"

Xenobio doesn't have to do this anyway, that's kinda the point. It's the equivalent of adding a chute beneath the surgery table in robotics to get rid of the leftover arms. If you're lazy, you just leave the random limbs lying there and it does no harm, just like leaving monkeys lying around.

Where's the creep? Most xenobiologists just leave the monkeys lying there unrecycled anyway, so all it does is make the room look less messy.

**Comment by cacogen** [comment]
oranges doesn't play and his response is coming from the point of view of the general sentiment that R&D is overpowered but click hunting is not intentional nor proper balance

plus magicarp have gotten a lot of nerfs too recently so you're slowly excising anything that makes this department fun anyway

**Comment by Fikou** [comment]
hugbox

**Comment by Ty-the-Smonk** [comment]
I don’t get why you would need this, it takes actually 2 seconds to click a button a few times to suck the, up

**Comment by kriskog** [comment]
Even as an upgrade, and especially that early in the techweb, this is stupid good. Anyone saying this isnt power creep needs to recalibrate their power creep sensors.

**Comment by ExcessiveUseOfCobblestone** [comment]
monkeys are recycled, linked to the recycler, so saying easy succ has 0 implications beyond qol is not true.

Whether it's good balance or not is not the point considering it didn't add a proper balance mechanic to replace it.

I'm not a fan of this, it's like 1 of 3 things you have to do as xenobio (the other being moving the slimes and placing the monkeys down). Same goes for it being a low-level tech upgrade.

**Comment by Dennok** [comment]
We has #49581 in perspective.

**Comment by Qustinnus** [comment]
That's not really a perspective if they're not linked prs at all. I'm not touching slime balance in that pr and it's not merged either.

**Comment by cacogen** [comment]
@dennok

what about making it so using a shortcut on a tile will look for dead monkeys and recycle the first one it finds instead? removes the click hunting while retaining similar balance

**Comment by Rasselbok** [comment]
> Even as an upgrade, and especially that early in the techweb, this is stupid good. Anyone saying this isnt power creep needs to recalibrate their power creep sensors.
> 
> When are we adding automatic feeding? Thats a "click hunt" too.

It's useful to people who are really new to xenobio and don't know how to properly get monkey cubes, I suppose - but the thing is, recycling is almost useless before you get the monkey recycler upgraded (0.2 monkeys per recycle), and by the time you get t4 (0.8 monkeys per recycle) you already have an industrial grey, which is roughly 100+ monkeys for every 4 clicks (draw plasma, inject 3 times), meaning you never need to recycle a monkey ever again except just for cleanliness.

You can get an industrial grey in like 15 minutes with a little work.

**Comment by Qustinnus** [comment]
so you're saying you dont need the QoL

**Comment by Rasselbok** [comment]
> so you're saying you dont need the QoL

I'm saying it's purely quality of life - it's there to get monkey corpses out of the way and not have corpses and fart gas in the room, but it doesn't actually directly affect the main gameplay loop.

I could see the argument that it's unnecessary, or doesn't really fit with the theme, or feature creep, but it's not /power creep/.

**Comment by Qustinnus** [comment]
you can literaly just click the button and prevent the same issue lol

**Comment by Rasselbok** [comment]
> you can literaly just click the button and prevent the same issue lol

yea, that's a better argument, it was just the whole idea of it somehow being 'too powerful' that i was arguing against.

**Comment by Qustinnus** [comment]
It's not too powerful, power creep means you keep buffing a certain gameplay element. The slime console was QoL and now this is QoL. Soon it'll be automatic monkey feeding and that'll also be QoL.

**Comment by optimumtact** [comment]
looks like the consensus is no

**Comment by cacogen** [comment]
>LGTM now dennok

**Comment by optimumtact** [comment]
if only we lived in a dictatorship

**Comment by cacogen** [comment]
WHAT

**Comment by Dennok** [comment]
> It's not too powerful, power creep means you keep buffing a certain gameplay element. The slime console was QoL and now this is QoL. Soon it'll be automatic monkey feeding and that'll also be QoL.

> automatic monkey feeding 
You already can drop 50 monkeys to chamber and forgot about feeding.
Or more with no console.

Or use grey crossbreeds to completely forgot about feeding.

**Comment by Dennok** [comment]
> what about making it so using a shortcut on a tile will look for dead monkeys and recycle the first one it finds instead? removes the pixel hunting while retaining similar balance

Maybe i make it remove dead monkeys on tile before placing new.
And maybe rearrange click options

**Comment by ExcessiveUseOfCobblestone** [comment]
if you spam monkeys doesn't it risk befriending the monkeys?

**Comment by Dennok** [comment]
> if you spam monkeys doesn't it risk befriending the monkeys?

Zero risk. If something you can close blastdoors of cell.

**Pull request closed without merging**